src/util/auth/hs_auth.ts

```diff
@@ -71,28 +61,22 @@ export class RequestAuthentication {
   };
 
   public checkLoggedIn = (req: Request, res: Response, next: NextFunction): void => {
-    passport.authenticate(
-      "cookie",
-      {
-        session: false
-      },
-      (err, user, info) => {
-        if (err) {
-          return next(err);
-        }
+    passport.authenticate("cookie", { session: false }, (err, user, info) => {
+      if (err) {
+        return next(err);
+      }
 
-        // There is not authenticated user, so redirect to logins
-        if (!user) {
-          const queryParam: string = querystring.encode({
-            returnto: `${process.env.APPLICATION_URL}${req.originalUrl}`
-          });
-          res.redirect(`${process.env.AUTH_URL}/login?${queryParam}`);
-          return;
-        }
-        res.locals.authLevel = user.authLevel;
-        return next();
+      // There is not authenticated user, so redirect to logins
+      if (!user) {
+        const queryParam: string = querystring.encode({
+          returnto: `${process.env.APPLICATION_URL}${req.originalUrl}`
+        });
+        res.redirect(`${process.env.AUTH_URL}/login?${queryParam}`);
+        return next("User not authenticated");
```

Comment:
I'm not 100% I'm right in saying this, but since the user is redirected, is calling next necessary?

Reply:
You are correct in that, we don't need to call the next function since they are already redirected. However, during testing I found that if the `next` callback function was not called here then the test would timeout.

If you look at test/unit/util/auth/auth.test.ts (test starting on line 174) I mocked the next() callback function, this was the only way I could see to test the `checkLoggedIn` function since passport uses callbacks instead of promises. Based on [this SO question](https://stackoverflow.com/q/55328833) I should look using promisfy with passport which might simplify the tests as well as removing the unnecessary `next` call.